### PR TITLE
feat: integrate openapi generator

### PR DESCRIPTION
**Pull request by tienan92it**
#### What's this PR do?

- [x] feat: integrate openapi generator
- [x] chore: add spectral rules for swagger validator
- [x] chore: init mock service workers
- [x] chore: remove swagger-typescript-api
- [x] chore: update generator api docs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,12 @@
     "@commitlint/config-conventional": "^17.3.0",
     "@dwarvesf/react-eslint-config": "^1.0.1",
     "@prettier/plugin-xml": "^3.2.0",
+    "@babel/core": "^7.22.5",
+    "@babel/helper-plugin-utils": "^7.20.2",
+    "@babel/preset-react": "^7.22.5",
```
Comment (zlatanpham) [inline package.json:53]: Do we still need those 3 packages? I think they have been removed in https://github.com/dwarvesf/nextjs-boilerplate/commit/7898e9d72b93d1a3e88ecc4814bff8fbddfd834c

File: .env.development [hunk 2]
```diff
@@ -1 +1,2 @@
-NEXT_PUBLIC_BASE_URL='/api'
\ No newline at end of file
+NEXT_PUBLIC_BASE_URL='https://hp-api.fly.dev/api/v1'
```
Comment (zlatanpham) [inline .env.development:1]: Do we have our branding URL?

File: .spectral.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
+const ibmCloudValidationRules = require('@ibm-cloud/openapi-ruleset') // Note 1
```
Comment (zlatanpham) [inline .spectral.js:1]: What does `// Note 1` mean?

File: src/pages/data-fetching.tsx [hunk 4]
```diff
@@ -3,12 +3,12 @@ import { Heading } from 'components/Heading'
 import { Card } from 'components/Card'
 import { Text } from 'components/Text'
 import { Table } from 'components/Table'
-import { User } from 'types/schema'
 import { Badge } from 'components/Badge'
-import { useFetchUsers } from 'hooks/data/useFetchUsers'
+import { useGetUsersList, User } from 'api'
```
Comment (zlatanpham) [inline src/pages/data-fetching.tsx:7]: Looks like we need to update the env variable in production?

image

**Review by zlatanpham — COMMENTED** [review]
Left a couple of questions

File: .spectral.js (hunk 3 above)
Comment (chinhld12) [reply .spectral.js:1]: Should be commented as:
```
// Extend default rules from the IBM OpenAPI Ruleset
```

File: docs/GETTING_STARTED.md [hunk 5]
```diff
@@ -41,7 +42,9 @@ to see your project.
 │   └── tests
 │── cypress                      # Cypress configuration and tests
 ├── tailwind.config.js           # Tailwind CSS configuration
-└── tsconfig.json                # TypeScript configuration
+├── tsconfig.json                # TypeScript configuration
+├── .spectral.js                 # Orval validator rules for Swagger
```
Comment (chinhld12) [inline docs/GETTING_STARTED.md:46]:
```suggestion
├── .spectral.js                 # IBM OpenAPI Ruleset for Orval use on validator rules for Swagger
```

File: src/api/mutator/requester.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      Authorization: `Bearer ${accessToken}`,
+    }
+  }
+  const promise = AXIOS_INSTANCE({ ...config, cancelToken: source.token }).then(
+    ({ data }) => data,
+  )
+
+  // @ts-ignore
+  promise.cancel = () => {
+    source.cancel('Query was cancelled by React Query')
```
Comment (ngolapnguyen) [inline src/api/mutator/requester.ts:26]: We are using `swr` right?

File: .env.development (hunk 2 above)
Comment (tienan92it) [reply .env.development:1]: No. Just waiting a proper url from BE boilerplate and I will replace it later. cc: @hieuphq

File: src/api/mutator/requester.ts (hunk 6 above)
Comment (tienan92it) [reply src/api/mutator/requester.ts:26]: Copied from `swr` example of Orval. Maybe a mistake or a normal term but I was confused too lol. It should be a general word like Axios or requester.

File: package.json (hunk 1 above)
Comment (tienan92it) [reply package.json]: seems like it's a conflict while I merged code. Just removed

**Review by zlatanpham — APPROVED** [review]

**Pull request merged**